dvc/command/plots.py

```diff
@@ -34,8 +50,15 @@ def run(self):
                     "you can only specify one target for `--show-vega`"
                 )
                 return 1
+            if self.args.json:
+                logger.error(
+                    "'--show-vega' and '--json' are mutually exclusive "
```

Comment:
[Q] Why are you using `--json` instead of `--show-json`?

Reply:
We have deprecated `--show-json` for simpler `--json` command. We may remove these `--show-*` options in 3.0.

Reply:
Till then, we should also provide `--show-json` as an alias.

https://github.com/iterative/dvc/blob/9803db6a64cd61352804c41accdd1a404971af13/dvc/command/metrics.py#L140-L142

@pared, also please add a test here if you add support for it: 
https://github.com/iterative/dvc/blob/9803db6a64cd61352804c41accdd1a404971af13/tests/unit/command/test_compat_flag.py#L35

Reply:
@skshetry done and done

Reply:
Thanks @pared.